### PR TITLE
Trilinos/Allowing empty partitions

### PR DESCRIPTION
**Pull request by roigcarlo**
Fixing some problems with MPI in kratos.

### CODE REVIEW

File: applications/TrilinosApplication/custom_strategies/convergencecriterias/trilinos_up_criteria.h [hunk 1]
```diff
@@ -202,9 +202,16 @@ class TrilinosUPCriteria : public ConvergenceCriteria< TSparseSpace, TDenseSpace
 
             int converged = 0;
 
+            double local_dimension = 0;
+            double dimension = 0;
+
+            if (r_model_part.GetCommunicator().LocalMesh().NumberOfElements())
```
Comment (philbucher) [inline applications/TrilinosApplication/custom_strategies/convergencecriterias/trilinos_up_criteria.h]:
```suggestion
            if (r_model_part.GetCommunicator().LocalMesh().NumberOfElements() > 0)
```

File: applications/TrilinosApplication/custom_strategies/convergencecriterias/trilinos_up_criteria.h [hunk 2]
```diff
@@ -202,9 +202,16 @@ class TrilinosUPCriteria : public ConvergenceCriteria< TSparseSpace, TDenseSpace
```
Comment (philbucher) [inline applications/TrilinosApplication/custom_strategies/convergencecriterias/trilinos_up_criteria.h:202]: man we should finally unify the ConvCrits (this one does basically the same as the DisplacementCriteria) and remove it, it is really not nicely implemented

Comment (roigcarlo) [reply applications/TrilinosApplication/custom_strategies/convergencecriterias/trilinos_up_criteria.h:202]: 100% Agree

**Review by philbucher — APPROVED** [review]
looks ok to me

@rubenzorrilla @RiccardoRossi any comment?

**Review by RiccardoRossi — APPROVED** [review]

**Pull request merged**